core/src/main/java/org/apache/iceberg/rest/ResourcePaths.java

```diff
@@ -40,39 +46,48 @@ public static String tokens() {
   }
 
   private final String prefix;
+  private final String namespaceSeparator;
 
   public ResourcePaths(String prefix) {
+    this(prefix, RESTUtil.NAMESPACE_ESCAPED_SEPARATOR);
+  }
+
+  private ResourcePaths(String prefix, String namespaceSeparator) {
     this.prefix = prefix;
+    this.namespaceSeparator = namespaceSeparator;
   }
 
   public String namespaces() {
     return SLASH.join("v1", prefix, "namespaces");
   }
 
   public String namespace(Namespace ns) {
-    return SLASH.join("v1", prefix, "namespaces", RESTUtil.encodeNamespace(ns));
+    return SLASH.join("v1", prefix, "namespaces", RESTUtil.encodeNamespace(ns, namespaceSeparator));
```

Comment:
Nit: I think it would be cleaner to add a method like `pathEncode` that applies the separator. Embedding this everywhere is causing lines to get too long.

Reply:
done